src/Editor/ImGuizmo/imguizmo.cpp

```diff
@@ -0,0 +1,1719 @@
+            dirPlaneY *= gContext.mAxisFactor[(axisIndex + 1) % 3];
+        }
+        else
+        {
+            vec_t dirPlaneNormalWorld;
+            dirPlaneNormalWorld.TransformVector(directionUnary[planNormal], gContext.mModel);
+            dirPlaneNormalWorld.Normalize();
+
+            vec_t dirPlaneXWorld(dirPlaneX);
+            dirPlaneXWorld.TransformVector(gContext.mModel);
+            dirPlaneXWorld.Normalize();
+
+            vec_t dirPlaneYWorld(dirPlaneY);
+            dirPlaneYWorld.TransformVector(gContext.mModel);
+            dirPlaneYWorld.Normalize();
+
+            vec_t cameraEyeToGizmo = Normalized(gContext.mModel.v.position - gContext.mCameraEye);
+            float dotCameraDirX = cameraEyeToGizmo.Dot3(dirPlaneXWorld);
+            float dotCameraDirY = cameraEyeToGizmo.Dot3(dirPlaneYWorld);
+
+            // compute factor values
+            float mulAxisX = (dotCameraDirX > 0.f) ? -1.f : 1.f;
+            float mulAxisY = (dotCameraDirY > 0.f) ? -1.f : 1.f;
+            dirPlaneX *= mulAxisX;
+            dirPlaneY *= mulAxisY;
+
+            belowAxisLimit = fabsf(dotCameraDirX) < angleLimit;
+            belowPlaneLimit = (fabsf(cameraEyeToGizmo.Dot3(dirPlaneNormalWorld)) > planeLimit);
+
+            // and store values
+            gContext.mAxisFactor[axisIndex] = mulAxisX;
+            gContext.mAxisFactor[(axisIndex + 1) % 3] = mulAxisY;
+            gContext.mBelowAxisLimit[axisIndex] = belowAxisLimit;
+            gContext.mBelowPlaneLimit[axisIndex] = belowPlaneLimit;
+        }
+    }
+
+    static void ComputeSnap(float*value, float snap)
+    {
+        if (snap <= FLT_EPSILON)
+            return;
+        float modulo = fmodf(*value, snap);
+        float moduloRatio = fabsf(modulo) / snap;
+        if (moduloRatio < snapTension)
+            *value -= modulo;
+        else if (moduloRatio >(1.f - snapTension))
+            *value = *value - modulo + snap * ((*value<0.f) ? -1.f : 1.f);
+    }
+    static void ComputeSnap(vec_t& value, float *snap)
+    {
+        for (int i = 0; i < 3; i++)
+        {
+            ComputeSnap(&value[i], snap[i]);
+        }
+    }
+
+    static float ComputeAngleOnPlan()
+    {
+        const float len = IntersectRayPlane(gContext.mRayOrigin, gContext.mRayVector, gContext.mTranslationPlan);
+        vec_t localPos = Normalized(gContext.mRayOrigin + gContext.mRayVector * len - gContext.mModel.v.position);
+
+        vec_t perpendicularVector;
+        perpendicularVector.Cross(gContext.mRotationVectorSource, gContext.mTranslationPlan);
+        perpendicularVector.Normalize();
+        float acosAngle = Clamp(Dot(localPos, gContext.mRotationVectorSource), -0.9999f, 0.9999f);
+        float angle = acosf(acosAngle);
+        angle *= (Dot(localPos, perpendicularVector) < 0.f) ? 1.f : -1.f;
+        return angle;
+    }
+
+    static void DrawRotationGizmo(int type)
+    {
+        ImDrawList* drawList = gContext.mDrawList;
+        ImGuiIO& io = ImGui::GetIO();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Variable 'io' is assigned a value that is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)